TLM/TLM/Manager/Impl/LaneArrowManager.cs

```diff
@@ -218,5 +220,238 @@ public class LaneArrowManager
         public static ICustomDataManager<List<Configuration.LaneArrowData>> AsLaneArrowsDM() {
             return Instance;
         }
+
+        public static class SeparateTurningLanes{
+            /// <summary>
+            /// separates turning lanes for all segments attached to nodeId
+            /// </summary>
+            public static void SeparateNode(ushort nodeId) {
+                NetNode node = Singleton<NetManager>.instance.m_nodes.m_buffer[nodeId];
+                if (nodeId == 0) {
+                    return;
+                }
+                if ((node.m_flags & NetNode.Flags.Created) == NetNode.Flags.None) {
+                    return;
+                }
+                //var nSegments = node.CountSegments();
+
+                for (int i = 0; i < 8; i++) {
+                    ushort segmentId = Singleton<NetManager>.instance.m_nodes.m_buffer[nodeId].GetSegment(i);
+                    if (segmentId == 0) {
+                        continue;
+                    }
+
+                    SeparateSegmentLanes(segmentId, nodeId);
+                }
+            }
+
+            /// <summary>
+            /// separates turning lanes for the input segment on the input node.
+            /// </summary>
+            public static void SeparateSegmentLanes(ushort segmentId, ushort nodeId) {
+                ref NetSegment seg = ref Singleton<NetManager>.instance.m_segments.m_buffer[segmentId];
+                bool startNode = seg.m_startNode == nodeId;
+
+                //list of outgoing lanes from current segment to current node.
+                IList<LanePos> laneList =
+                    Constants.ServiceFactory.NetService.GetSortedLanes(
+                        segmentId,
+                        ref Singleton<NetManager>.instance.m_segments.m_buffer[segmentId],
+                        startNode,
+                        LaneArrowManager.LANE_TYPES,
+                        LaneArrowManager.VEHICLE_TYPES,
+                        true
+                        );
+                int srcLaneCount = laneList.Count();
+                if (srcLaneCount == 1)
+                    return;
+
+                int leftLanesCount = CountTargetLanesTowardDirection(segmentId, nodeId, ArrowDirection.Left);
+                int rightLanesCount = CountTargetLanesTowardDirection(segmentId, nodeId, ArrowDirection.Right);
+                int forwardLanesCount = CountTargetLanesTowardDirection(segmentId, nodeId, ArrowDirection.Forward);
+                int totalLaneCount = leftLanesCount + forwardLanesCount + rightLanesCount;
+                int numdirs = Convert.ToInt32(leftLanesCount > 0) + Convert.ToInt32(rightLanesCount > 0) + Convert.ToInt32(forwardLanesCount > 0);
+
+                Log._Debug($"SeparateSegmentLanes: totalLaneCount {totalLaneCount} | numdirs = {numdirs} | outgoingLaneCount = {srcLaneCount}");
+
+                if (numdirs < 2)
+                    return; // no junction
```

Comment:
this if statement needs curly braces

Reply:
To be style consistent across whole project, yes